src/django_components/templatetags/component_tags.py

```diff
@@ -809,164 +815,279 @@ class ParsedTag(NamedTuple):
     parse_body: Callable[[], NodeList]
 
 
+class TagArg(NamedTuple):
+    name: str
+    positional_only: bool
+
+
 def _parse_tag(
     parser: Parser,
     token: Token,
     tag_spec: TagSpec,
+    tag_id: str,
 ) -> ParsedTag:
-    # Use a unique ID to be able to tie the fill nodes with components and slots
-    # NOTE: MUST be called BEFORE `parse_body()` to ensure predictable numbering
-    tag_id = gen_id()
+    tag_name, raw_args, raw_kwargs, raw_flags, is_inline = _parse_tag_preprocess(parser, token, tag_spec)
+
+    parsed_tag = _parse_tag_process(
```

Comment:
Here, after `_parse_tag_preprocess()` and before `_parse_tag_process()`, is where one of the plugin's hooks would go, in order to be able to e.g. add extra args / kwargs.